pkg/cloud/aws/aws.go

```diff
@@ -2,36 +2,85 @@ package aws
 
 import (
 	"embed"
+	"fmt"
 
-	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/common"
+	"github.com/asaskevich/govalidator"
 	appsv1 "k8s.io/api/apps/v1"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/common"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/config"
 )
 
+const providerName = "aws"
+
 var (
 	//go:embed assets/*
-	awsFS embed.FS
-
-	awsResources []client.Object
+	assetsFs embed.FS
 
-	awsSources = []common.ObjectSource{
-		{Object: &appsv1.Deployment{}, Path: "assets/deployment.yaml"},
+	templates = []common.TemplateSource{
+		{ReferenceObject: &appsv1.Deployment{}, EmbedFsPath: "assets/deployment.yaml"},
 	}
 )
 
-func init() {
-	var err error
-	awsResources, err = common.ReadResources(awsFS, awsSources)
-	utilruntime.Must(err)
+type imagesReference struct {
+	CloudControllerManager string `valid:"required"`
+}
+
+var templateValuesValidationMap = map[string]interface{}{
+	"images": "required",
+}
+
+type awsAssets struct {
+	operatorConfig config.OperatorConfig
+	templates      []common.ObjectTemplate
+	images         *imagesReference
 }
 
-// GetResources returns a list of AWS resources for provisioning CCM in running cluster
-func GetResources() []client.Object {
-	resources := make([]client.Object, len(awsResources))
-	for i := range awsResources {
-		resources[i] = awsResources[i].DeepCopyObject().(client.Object)
+func (a *awsAssets) GetRenderedResources() ([]client.Object, error) {
+	renderedObjects, err := common.RenderTemplates(a)
+	if err != nil {
+		return nil, err
 	}
+	return renderedObjects, nil
+}
+
+func (a *awsAssets) GetTemplates() []common.ObjectTemplate {
+	return a.templates
+}
 
-	return resources
+func (a *awsAssets) GetTemplateValues() (common.TemplateValues, error) {
+	values := common.TemplateValues{
+		"images": a.images,
+	}
+	_, err := govalidator.ValidateMap(values, templateValuesValidationMap)
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
+}
+
+func (a awsAssets) GetOperatorConfig() config.OperatorConfig {
+	return a.operatorConfig
+}
+
+func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets, error) {
+	images := &imagesReference{
+		CloudControllerManager: config.ImagesReference.CloudControllerManagerAWS,
+	}
+
+	_, err := govalidator.ValidateStruct(images)
```

Comment:
```suggestion 
if _, err := govalidator.ValidateStruct(images); err != nil {
```

Reply:
Bit opinionated but I don't like such construction here. Sorry, wont fix it :) The same for other comments around errors inlining for validation.